summingbird-storm/src/main/scala/com/twitter/summingbird/storm/SpoutProvider.scala

```diff
@@ -68,64 +60,48 @@ case class SpoutProvider(storm: Storm, stormDag: Dag[Storm], node: StormNode, jo
     }
   }
 
-  private def getCountersForJob(jobId: JobId): Seq[(Group, Name)] = JobCounters.getCountersForJob(jobId).getOrElse(Nil)
-
-  /**
-   * This function takes the metrics and counters to register to the TopologyContext of a JobID and outputs a function which can be called on openHook.
-   */
-  private def createRegistrationFunction(
-      metrics: SpoutStormMetrics,
-      counters: Seq[(Group, Name)],
-      jobID: JobId): Externalizer[(TopologyContext) => Unit] = {
-
-    Externalizer({ context: TopologyContext =>
-      // Register metrics passed in SpoutStormMetrics option.
-      metrics.metrics().foreach {
-        x: StormMetric[IMetric] =>
-          context.registerMetric(x.name, x.metric, x.interval.inSeconds)
-      }
-      // Register summingbird counter metrics.
-      StormStatProvider.registerMetrics(jobID, context, counters)
-      SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
-    })
-  }
-
-
-  private def counter(jobID: JobId, nodeName: Group, counterName: Name) = new Counter(Group("summingbird." + nodeName.getString), counterName)(jobID) with Incrementor
-
-
   /*
    * This method formats the tuples to batched format and wraps the spout into KeyValueSpout.
    */
   private def createSpoutToFeedSummer[K, V](
-      stormDag: Dag[Storm],
-      node: StormNode,
-      builder: SummerBuilder,
-      spout: Spout[(Timestamp, (K, V))],
-      fn: Externalizer[(TopologyContext) => Unit],
-      flushExecTimeCounter: Incrementor,
-      executeTimeCounter: Incrementor): IRichSpout = {
+    stormDag: Dag[Storm],
+    node: StormNode,
+    builder: SummerBuilder,
+    spout: Spout[(Timestamp, (K, V))],
+    fn: Externalizer[(TopologyContext) => Unit],
+    flushExecTimeCounter: Incrementor,
+    executeTimeCounter: Incrementor): IRichSpout = {
 
     val summerParalellism = getOrElse(stormDag, node, Constants.DEFAULT_SUMMER_PARALLELISM)
     val summerBatchMultiplier = getOrElse(stormDag, node, Constants.DEFAULT_SUMMER_BATCH_MULTIPLIER)
     val keyValueShards = executor.KeyValueShards(summerParalellism.parHint * summerBatchMultiplier.get)
     val summerProducer = node.members.collect { case s: Summer[_, _, _] => s }.head.asInstanceOf[Summer[Storm, K, V]]
     val batcher = summerProducer.store.mergeableBatcher
-    val kvinj = new KeyValueInjection[(K, BatchID), (Timestamp, V)]
     val formattedSummerSpout = spout.map {
-      case (time, (k: K, v: V)) => kvinj((k, batcher.batchOf(time)), (time, v))
+      case (time, (k: K, v: V)) => ((k, batcher.batchOf(time)), (time, v))
```

Comment:
let's remove `(k: K, v: V)` and use `(k, v)`. In a case statement, `:K` means match on type, which is not possible when `K` is generic, so can lead to warnings.